src/main/java/org/onebusaway/alexa/lib/ObaUserClient.java

```diff
@@ -35,18 +37,37 @@
 public class ObaUserClient {
     private static final int DEFAULT_SEARCH_RADIUS_METERS = 40000;
 
-    public ObaUserClient(String endpoint) {
-        log.debug("Instantiating ObaUserClient with endpoint " + endpoint);
+    public ObaUserClient(String obaBaseUrl) {
+        log.debug("Instantiating ObaUserClient with obaBaseUrl " + obaBaseUrl);
         try {
-            ObaApi.getDefaultContext().setBaseUrl(
-                    String.format("http://api.%s.onebusaway.org/", endpoint));
+            ObaApi.getDefaultContext().setBaseUrl(obaBaseUrl);
         } catch (URISyntaxException e) {
-            log.error("OBA Endpoint we constructed was invalid: " + e.getMessage());
+            log.error("ObaBaseUrl we constructed was invalid: " + e.getMessage());
         } catch (Exception e) {
-            log.error("Exception setting OBA base URL: " + e.getMessage());
+            log.error("Exception setting ObaBaseUrl: " + e.getMessage());
             throw e;
         }
     }
+
+    /**
+     * Get the closest region to a given location.  Regions come from the Regions API
+     * (http://regions.onebusaway.org/regions-v3.json), which is a centralized directory of all known
+     * OneBusAway servers.  Each region has it's own OBA REST API endpoint
+     * (http://developer.onebusaway.org/modules/onebusaway-application-modules/current/api/where/index.html),
+     * which we will then use to get stop and arrival data for that region.
+     *
+     * @param l geographic location used to search for nearby regions
+     * @return the closest region in the Regions API to the given location, or null if there are no nearby regions
+     * (within 100 miles of the provided location) or a region couldn't be found.
+     */
+    public static ObaRegion getClosestRegion(Optional<Location> l) {
```

Comment:
Since this function requires a non-null `Location` argument (Optional.empty is not supported), it should take `Location` directly, leaving to the caller the matter of what should happen if there's no location.